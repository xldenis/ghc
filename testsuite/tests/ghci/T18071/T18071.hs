{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE PolyKinds #-}
{-# LANGUAGE QuantifiedConstraints #-}
{-# LANGUAGE UndecidableInstances #-}
{-# LANGUAGE FlexibleContexts #-}
module Bug where

import Data.Kind
import Data.Proxy

class    (forall (z :: k). Show (Proxy z)) => ShowProxy (k :: Type) where
instance (forall (z :: k). Show (Proxy z)) => ShowProxy (k :: Type)

data T :: Type -> Type

data U = A | B
