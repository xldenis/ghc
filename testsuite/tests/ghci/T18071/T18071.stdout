instance [safe] ShowProxy U -- Defined at T18071.hs:12:10
